src/main/java/org/mskcc/cbio/oncokb/web/rest/CronJobController.java

```diff
@@ -182,6 +205,97 @@ public void checkTrialAccounts() {
         mailService.sendTrialAccountExpiresMail(DAYS_TO_CHECK, userDTOS);
     }
 
+    @GetMapping(path = "/exposed-tokens")
+    public void exposedTokensInfo() throws IOException, InterruptedException {
+        List<Token> tokens = tokenService.findAll();
+        log.info("Searching exposed tokens pipeline begins...");
+        List<ExposedToken> results = new ArrayList<>();
+        for (Token token : tokens){
+            String q = token.getToken().toString();
+
+            int githubCount = 0;
+            try {
+                GitHub github = new GitHubBuilder().withOAuthToken(applicationProperties.getGithubToken()).build();
+                List<GHContent> gitRes = github.searchContent().q(q).list().toList();
+                githubCount = gitRes.size();
+                if (githubCount == 1 && gitRes.get(0).getHtmlUrl().indexOf("https://github.com/knowledgesystems/oncokb-deployment/") == 0){
+                        githubCount = 0;
+                }
+            }
+            catch (Exception e) {
+                e.printStackTrace();
+            }
+
+            int googleCount = 0;
+            try{
+                HttpEntity entity = getResponse(q).getEntity();
+                Document googleDoc = Jsoup.parse(EntityUtils.toString(entity));
+                Elements resultEle = googleDoc.select("h3");
+                googleCount = resultEle.size();
+                if (googleCount > 0) {
+                    String description = googleDoc.select(".kCrYT > div > div > div > div > div").text();
+                    if (description.indexOf(q) == -1)
+                        googleCount = 0;
+                }
+            }
+            catch (Exception e){
+                e.printStackTrace();
+            }
+            
+            int baiduCount = 0;
+            try{
+                Document document = Jsoup.connect(String.format("https://www.baidu.com/s?wd=\"%s\"", q)).get();
+                Elements elements = document.select("div.result.c-container");
+                baiduCount = elements.size();
+            }
+            catch (Exception e){
+                e.printStackTrace();
+            }
+           
+
+            if (githubCount > 0 || googleCount > 0 || baiduCount > 0){
+                ExposedToken t = new ExposedToken();
+                UserDTO user = userMapper.userToUserDTO(token.getUser());
+                t.setToken(q);
+                t.setEmail(user.getEmail());
+                t.setFirstName(user.getFirstName());
+                t.setLastName(user.getLastName());
+                t.setLicenseType(user.getLicenseType() != null ? user.getLicenseType().name() : "");
+                String source = "";
+                if(githubCount > 0){
```

Comment:
This is optional, I would just do a List and do a collection joining later.

Reply:
Could you please show me how to do it here? I've tried but I cannot implement same result by using such process. Appreciate!

Reply:
@Leowisd do a ArrayList, then add element if applicable. Then do a .stream().collect(Collectors.joining(" "))

Reply:
Oh, I got it. I misunderstood your first comment. Sorry about that.